instrumentation/net_http/lib/opentelemetry/instrumentation/net/http/patches/instrumentation.rb

```diff
@@ -67,6 +84,8 @@ def annotate_span_with_response!(span, response)
 
               span.set_attribute(OpenTelemetry::SemanticConventions::Trace::HTTP_STATUS_CODE, status_code)
               span.status = OpenTelemetry::Trace::Status.error unless (100..399).include?(status_code.to_i)
+
+              safe_execute_hook(instrumentation_config[:response_hook], span, response)
```

Comment:
Same deal here — maybe `safe_execute_hook if instrumentation_config[:response_hook]`?

Reply:
I'm checking the condition inside the function, so I preferred to not to do it twice and keep the method safer. I don't mind changing it though, if you prefer.

Reply:
I would rather check the condition prior to entering the method 😄

Reply:
Done